lib/vrl/stdlib/src/to_regex.rs

```diff
@@ -0,0 +1,76 @@
+use regex::Regex;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToRegex;
+
+impl Function for ToRegex {
+    fn identifier(&self) -> &'static str {
+        "to_regex"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::bytes,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "regex",
+                source: "to_regex(s'foo')",
```

Comment:
Could we provide an example showing a pattern? For example: `to_regex(s'^foobar$')`.